xarray/plot/plot.py

```diff
@@ -359,16 +359,16 @@ def step(darray, *args, where="pre", linestyle=None, ls=None, **kwargs):
     if where not in {"pre", "post", "mid"}:
         raise ValueError("'where' argument to step must be " "'pre', 'post' or 'mid'")
 
-    if ls is not None:
-        if linestyle is None:
-            linestyle = ls
+    if ds is not None:
```

Comment:
raise an error if `linestyle` or `ls` is in `kwargs`?

Reply:
`linestyle` and `ls` are still valid things to pass; you just can't mix draw styles into the string.